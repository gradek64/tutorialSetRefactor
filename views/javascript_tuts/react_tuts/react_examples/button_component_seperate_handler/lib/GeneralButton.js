"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _react = require("react");

var _react2 = _interopRequireDefault(_react);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var GeneralButton = function GeneralButton(props) {
  var callParent = props.callParent,
      label = props.label;

  return _react2.default.createElement(
    "button",
    { onClick: function onClick() {
        return callParent(label);
      } },
    label
  );
};

exports.default = GeneralButton;