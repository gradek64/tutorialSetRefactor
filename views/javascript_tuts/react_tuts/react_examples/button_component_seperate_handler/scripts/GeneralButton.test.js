import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import GeneralButton from './GeneralButton';


describe('GeneralButton', () => {
    it('should render correctly in "debug" mode', () => {
        const component = renderer
            .create(<GeneralButton  />)
            .toJSON();

        console.log(component);
  
        expect(component).toMatchSnapshot();
    });
    it('should handle the click event', () => {
        const myMock = jest.fn(x => 42 + x);
        const wrapper = shallow(<GeneralButton />);
        //pass method to props
        wrapper.setProps({ callParent: ()=>{myMock(5);} });
        wrapper.find('button').simulate('click');
        expect(myMock.mock.results[0].value).toBe(47);
    });
});