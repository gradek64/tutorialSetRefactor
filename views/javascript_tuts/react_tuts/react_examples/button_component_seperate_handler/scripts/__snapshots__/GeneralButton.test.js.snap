// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeneralButton should render correctly in "debug" mode 1`] = `
<button
  onClick={[Function]}
/>
`;
